torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -205,7 +207,16 @@ Tensor norm_backward(Tensor grad, const Tensor& self, const optional<Scalar> & p
     self_scaled = self;
     scale_v = grad / norm;
   } else if (std::isinf(p)) {
-    Tensor is_eq_max = (self.abs() == norm).logical_or_(self.isnan().logical_and_(norm.isnan())).type_as(self);
+    Tensor self_and_norm_isnan;
+    const auto self_isnan = self.isnan();
+    const auto norm_isnan = norm.isnan();
+    if (!areAnyTensorSubclassLike({self, norm})) {
```

Comment:
could this be 
```cpp
auto self_and_norm_isnan = areAnyTensorSubclassLike({self, norm}) ?
    self_isnan.logical_and(norm_isnan) : 
    self_isnan.logical_and_(norm_isnan);
```
?

Reply:
That doesn't avoid the refcount: `auto` resolves to `Tensor` and never a reference type

Reply:
The way to avoid the refcount bump is to have a lot of code duplication like there was before, let me know if you would prefer that.

Reply:
I does avoid one refcount compared to the code above where you first create a (empty) Tensor, then assign to it.
We could also do `const auto&` here and rely on lifetime extension to keep the returned Tensor alive in case it is fresh?

Reply:
Today I learned about lifetime extension: https://abseil.io/tips/107

Reply:
> I does avoid one refcount compared to the code above where you first create a (empty) Tensor, then assign to it.

Oh I see, that's a great point. Thanks for the clarification